Documentation/api.md

```diff
@@ -760,18 +778,13 @@ StorageSpec defines the configured storage for a group Prometheus servers. If ne
 
 ## TLSConfig
 
-TLSConfig specifies TLS configuration parameters.
+TLSConfig extends the safe TLS configuration with file parameters.
 
 | Field | Description | Scheme | Required |
 | ----- | ----------- | ------ | -------- |
 | caFile | Path to the CA cert in the Prometheus container to use for the targets. | string | false |
-| ca | Stuct containing the CA cert to use for the targets. | [SecretOrConfigMap](#secretorconfigmap) | false |
 | certFile | Path to the client cert file in the Prometheus container for the targets. | string | false |
-| cert | Struct containing the client cert file for the targets. | [SecretOrConfigMap](#secretorconfigmap) | false |
 | keyFile | Path to the client key file in the Prometheus container for the targets. | string | false |
-| keySecret | Secret containing the client key file for the targets. | *[v1.SecretKeySelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#secretkeyselector-v1-core) | false |
-| serverName | Used to verify the hostname for the targets. | string | false |
-| insecureSkipVerify | Disable target certificate validation. | bool | false |
```

Comment:
I am sorry I think I overlooked the removal of those lines, my bad, but as far as I see this affects all other references to *TLSConfig, so we'd break very many things.

I believe we cannot simply break this API as is.

Reply:
Since there is no net change of fields here, I rather suggest to keep them in `TLSConfig`.

Reply:
Ah, I see some discussion already happened in https://github.com/prometheus-operator/prometheus-operator/pull/3370#discussion_r461666407.

Just raising awareness that we're changing a v1 API here, no?

cc @brancz

Reply:
Agreed, good catch Serg, we should not be changing this in this way, we need to make the transition smooth, as this is all v1 API and cannot be broken. This would be a major breaking change, that can be reserved for v2 but not right now.

Reply:
This seems more like a rendering issue? AFAICT the definition hasn't changed in the CRD.

Reply:
Seems like its inline now https://github.com/prometheus-operator/prometheus-operator/pull/3370/files#diff-c36225fd2ff42566c85e4a1b700bc100R954-R964 Seems like generation is broken then, as it removes the API references

Reply:
Oh wow, good catch. After looking a bit into it, it seems that the generation tool is intentionally removing the inlined fields from the documentation https://github.com/prometheus-operator/prometheus-operator/blob/master/cmd/po-docgen/api.go#L224-L226.

Reply:
The issue here is that, if I enable inlined fields printing in the generator we will also see the [TypeMeta](https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#TypeMeta) information from all the custom resources that we define in the API. This will add the Kind and the APIVersion of the objects to the documentation.
Is this something we want or should I enable the generation for all inlined fields except TypeMeta?

Reply:
> should I enable the generation for all inlined fields except TypeMeta?

this sounds like a good option to me.

Reply:
> Is this something we want or should I enable the generation for all inlined fields except TypeMeta?
if that is possible, then yes :+1:

Reply:
ok, I will create a seperate PR for that